src/App.js

```diff
@@ -18,9 +19,13 @@ function App() {
 
       <Router>
         <Switch>
-          <Route exact path="/" children={<SearchPage />} />
-          {/* <Route path="/cocktail/:id" children={<CocktailView />} /> */}
+          <Route path="/search/:query?" component={SearchPage} />
```

Comment:
J'utilise la propriété `component` au lieu de `Children`. De cette manière, React-Router donne des `props` à mon composants.

Plus précisément cela veut dire que je peux utiliser `this.props.history` ou `this.props.location` quand je suis dans le composant `SearchPage`.